doc/source/development/rfc/rfc94_field_precision_width_metadata.rst

```diff
@@ -0,0 +1,173 @@
+.. _rfc-94:
+
+=============================================================
+RFC 94: Numeric fields width/precision metadata
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2023-Feb-17
+Status:        Development
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+The document proposes and describes the introduction of a new method
+to the `OGRFieldDefn` class that returns information for numeric real
+fields about how the precision and width have been calculated by the
+driver and have to be interpreted.
+
+Motivation
+----------
+
+Applications (for example OGR/GDAL utils and QGIS) may require information
+about the field width and precision in order to convert between different
+formats or to validate user input on editing.
+
+Different drivers may calculate the width and precision differently by including
+or not the decimal separator and/or the minus sign into the calculated length.
+
+There is currently no way for an application to access this information and this
+can lead to loss of width or precision while converting between formats or when the
+application decides to stay on the safe side and reduce the width reported by GDAL.
+
+Additionally, attention must be paid to the meaning of "width" and "precision":
+OGR "width" corresponds to SQL "precision" and OGR "precision" corresponds to SQL "scale".
+
+
+For reference:
+
+- https://trac.osgeo.org/gdal/ticket/6960
+- https://issues.qgis.org/issues/11755
+- https://issues.qgis.org/issues/15188#note-8
+- https://github.com/qgis/QGIS/issues/51849
+
+
+Current drivers behavior
+------------------------
+
+Here is a list of the drivers and how they behave with respect to the width and precision,
+(for databases we refer to the `NUMERIC` data type).
+
+================== ================================== =====================
+ Driver             Width Includes Decimal Separator   Width Includes Sign
+================== ================================== =====================
+ Shapefile          YES                                YES
+ MapInfo            ?                                  ?
+ PostgreSQL         NO                                 NO
+ MySQL              NO                                 NO
+ MSSQL              NO                                 NO
+ OCI                NO                                 NO
+ GPKG               N/A                                N/A
+ CSV (from .csvt)   N/A                                N/A
+ HANA               ?                                  ?
+ FlatGeoBuf         ?                                  ?
+ FileGDB            ?                                  ?
```

Comment:
FileGDB is interesting. So looking at the code in ogr/ogrsf_frmts/filegdb/FGdbLayer.cpp, we currently ogr_width to the FileGDB <Field><Length> XML element and ogr_precision to <Field><Precision>. And we set <Field><Scale> to 0.
But there is https://desktop.arcgis.com/en/arcmap/latest/manage-data/tables/arcgis-field-data-types.htm#GUID-70728949-2016-4DB5-A723-5493B4330AE0 which mentions:
"Specifying the precision and scale allows you to restrict the range of values and number formats a field can accept, giving you greater control. For example, if you specify a float with a precision of 4 and a scale of 2, the field accepts 12.34 (or 12.3 or 12). If you enter 12.345 in the field, an error message displays, as this exceeds the maximum number of digits and decimal places allowed. On the other hand, if you specify a float with a precision of 5 and a scale of 3, the field allows you to enter 12.345, 12.34, or 12.3."

and https://resources.arcgis.com/en/help/arcobjects-java/api/arcobjects/com/esri/arcgis/geodatabase/Field.html#getPrecision():
"Precision is the number of digits in a number. For example, the number 56.78 has a precision of 4. Precision is only valid for fields that are numeric. Precision is always returned as 0 from personal or File geodatabase fields."

https://resources.arcgis.com/en/help/arcobjects-java/api/arcobjects/com/esri/arcgis/geodatabase/Field.html#getScale()
"Scale is the number of digits to the right of the decimal point in a number. For example, the number 56.78 has a scale of 2. Scale applies only to fields that are double. Scale is always returned as 0 from personal or File geodatabase fields."

and
https://resources.arcgis.com/en/help/arcobjects-java/api/arcobjects/com/esri/arcgis/geodatabase/Field.html#getLength()
"The maximum length, in bytes, for values described by the field. "

So it seems to me that the FileGDB driver needs to be corrected to not set <Precision> on writing, and ignore <Length> and <Precision> on reading for floating-point fields (<Length> for strings & binary fields is likely OK)
The OpenFileGDB one would just need to be corrected in its AlterFieldDefn() implementation to not propagate width and precision.

Reply:
Do I get it right that for FileGDB is `N/A` ?

Reply:
yes FileGDB is N/A